java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -640,7 +640,7 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_castTo(JNIEnv *env, jclas
     cudf::column_view *column = reinterpret_cast<cudf::column_view *>(handle);
     cudf::data_type n_data_type = cudf::jni::make_data_type(type, scale);
     std::unique_ptr<cudf::column> result;
-    if (n_data_type == column->type()) {
+    if (n_data_type.id() == column->type().id() && n_data_type.scale() == column->type().scale()) {
```

Comment:
Rather than making a copy in C++, Java could check for the degenerate cast and simply increment the refcount on the input column and return it.

Reply:
The change to this if-statement should not be required since there is an `operator==` function for `cudf::data_type` that already checks the `id` and the `scale` match:
https://github.com/rapidsai/cudf/blob/af411364164b3bce5c8fa6c0294c83ba2c9dfa50/cpp/include/cudf/types.hpp#L292-L296

Reply:
So 2 things.  First for some APIs we can and already do that.  But if it is a `ColumnView` instead of a `ColumnVector` there is no ref count yet so we cannot.

Second I already fixed this issue a whole ago. I plan on rerunning these tests without this one line of change to verify if my original fix really did work like I expected it to. c16a0a50beb

Reply:
Yup I manually verified that reverting this one line still has all of the tests pass.

Reply:
I have reverted this line. Thank you all for pointing out this :)